workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectDispatcherSystem.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Unity.Entities;
+
+namespace Improbable.Gdk.Core
+{
+    /// <summary>
+    ///     Gathers incoming dispatcher ops and invokes callbacks on relevant GameObjects.
+    /// </summary>
+    [UpdateInGroup(typeof(SpatialOSReceiveGroup.GameObjectReceiveGroup))]
+    public class GameObjectDispatcherSystem : ComponentSystem
+    {
+        public readonly Dictionary<int, SpatialOSBehaviourManager> EntityIndexToSpatialOSBehaviourManager = new Dictionary<int, SpatialOSBehaviourManager>();
+
+        public readonly HashSet<GameObjectComponentDispatcherBase> GameObjectComponentDispatchers =
+            new HashSet<GameObjectComponentDispatcherBase>();
+
+        protected override void OnCreateManager(int capacity)
+        {
+            base.OnCreateManager(capacity);
+
+            FindGameObjectComponentDispatchers();
+            GenerateComponentGroups();
+        }
+
+        private void FindGameObjectComponentDispatchers()
+        {
+            var gameObjectComponentDispatcherTypes = AppDomain.CurrentDomain.GetAssemblies()
+                .SelectMany(assembly => assembly.GetTypes())
+                .Where(type => typeof(GameObjectComponentDispatcherBase).IsAssignableFrom(type) && !type.IsAbstract).ToList();
+
+            foreach (var gameObjectComponentDispatcherType in gameObjectComponentDispatcherTypes)
```

Comment:
If you switch `GameObjectComponentDispatchers` to a List you can use LINQ:

```GameObjectComponentDispatchers = gameObjectComponentDispatcherTypes.Select(type => (GameObjectComponentDispatcherBase)Activator.CreateInstance(type)).ToList();```

Reply:
This is super elegant. I love it. I'm using `GameObjectComponentDispatchers.AddRange()` now because `GameObjectComponentDispatchers` can stay readonly that way.